sycl/test-e2e/Graph/RecordReplay/prefetch.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %{build} -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %{run} %t.out
+
+// CHECK: piextCommandBufferPrefetchUSM
```

Comment:
nit:

it might be possible to add another CHECK for the purpose of making sure that e.g. prefetchUSM is called before the kernel is executed in which the prefetched memory is used, for the purpose of checking that depends_on is working correctly (or if I read the spec right "make_edge").

Reply:
Good comment, made me realize that we didn't actually call FileCheck as part of the the `RUN:` line 🤦   Updated to call FileCheck and also verify in CHECKs that there's a dependency between the two commands by verifying that the `num_sync_points_in_wait_list` parameter to `piextCommandBufferNDRangeKernel` is 1.